.script/tests/detectionTemplateSchemaValidation/DetectionTemplateSchemaValidationTests.cs

```diff
@@ -10,19 +12,50 @@
 
 namespace Kqlvalidations.Tests
 {
-    public class DetectionTemplateStructureValidationTests
+    public class DetectionTemplateSchemaValidationTests
     {
         private static readonly string DetectionPath = DetectionsYamlFilesTestData.GetDetectionPath();
 
         [Theory]
         [ClassData(typeof(DetectionsYamlFilesTestData))]
         public void Validate_DetectionTemplates_HaveValidTemplateStructure(string detectionsYamlFileName)
+        {
+            Action<ScheduledTemplateInternalModel> validateConnectorSchema = (ScheduledTemplateInternalModel templateObject) => {
+                var validationContext = new ValidationContext(templateObject);
+                Validator.ValidateObject(templateObject, validationContext, true);
+            };
+
+            ValidateDetectionTemplate(detectionsYamlFileName, validateConnectorSchema);
+        }
+
+        [Theory]
+        [ClassData(typeof(DetectionsYamlFilesTestData))]
+        public void Validate_DetectionTemplates_HaveValidConnectorsIds(string detectionsYamlFileName)
+        {
+            Action<ScheduledTemplateInternalModel> validateConnectorsIds = (ScheduledTemplateInternalModel templateObject) => {
+                List<string> connectorIds = templateObject.RequiredDataConnectors
+                    .Where(requiredDataConnector => requiredDataConnector.ConnectorId != null)
+                    .Select(requiredDataConnector => requiredDataConnector.ConnectorId).ToList();
+
+                connectorIds.ForEach(connectorId => {
+                    if (!TemplatesSchemaValidationsReader.ValidConnectorIds.Contains(connectorId))
```

Comment:
do you want to have a whitelist until everything is fixed by mstic?

Reply:
no, this PR will be merged only after they will fix.